src/components/ModalWithForm.js

```diff
@@ -8,6 +8,15 @@ export default class ModalWithForm extends Modal {
     this._inputElements = Array.from(
       this.modalForm.querySelectorAll(".modal__input")
     );
+    this._modalSubmitButton = this._modalElement.querySelector(
+      ".modal__submit-button"
+    );
+    this._submitButtonText = this._modalSubmitButton.textContent;
+  }
+
+  resetFormAndDisableButton() {
+    this.modalForm.reset();
+    this._modalSubmitButton.disabled = true;
```

Comment:
* Please, make only method `resetForm` inside `ModalWithForm` because disabling is handled by `FormValidator`
* Now only class `FormValidator` should **`disable/enable`** submit buttons and remove errors from the inputs (if you do it)